roles/edpm_prepare/tasks/kustomize_and_deploy.yml

```diff
@@ -0,0 +1,84 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Perform kustomizations to the OpenStackControlPlane CR
+  vars:
+    cifmw_edpm_prepare_openstack_crs_path: >-
+      {{
+        [
+          cifmw_edpm_prepare_manifests_dir,
+          cifmw_install_yamls_defaults['NAMESPACE'],
+          'openstack',
+          'cr'
+        ] | ansible.builtin.path_join
+      }}
+  ci_kustomize:
+    target_path: "{{ cifmw_edpm_prepare_openstack_crs_path }}"
+    sort_ascending: false
+    kustomizations: "{{ cifmw_edpm_prepare_kustomizations }}"
+    kustomizations_paths: >-
+      {{
+        [
+          (
+            [
+              cifmw_edpm_prepare_manifests_dir,
+              'kustomizations',
+              'controlplane'
+            ] | ansible.builtin.path_join
+          )
+        ]
+      }}
+  register: cifmw_edpm_prepare_crs_kustomize_result
+
+- name: Log the CR that is about to be applied
+  ansible.builtin.debug:
+    var: cifmw_edpm_prepare_crs_kustomize_result
+
+- name: Apply the OpenStackControlPlane CR
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  cifmw.general.ci_script:
```

Comment:
i know this is a copy of what was there but just FYI we added 'extra_args' when removing ci_make so you can use that instead of environment

```
  extra_args:
    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
    PATH: "{{ cifmw_path }}"
```
environment: works fine, but we don't get this in the resulting logs or output from ci_script (see https://github.com/openstack-k8s-operators/ci-framework/blob/f957797867789b42ffdf6031945a844853d7ab4b/plugins/action/ci_script.py#L118)

I am in 2 minds about updating this here but up to you.

Reply:
yeah, it was just a code move from a file to another, so I didn't pay attention to improvements. I would say that we can make a hard this as hard requirement for new code. For this role, would be good to update in all places (at least for the role)